src/admin/components/elements/ReactSelect/index.tsx

```diff
@@ -41,8 +52,55 @@ const SelectAdapter: React.FC<ReactSelectAdapterProps> = (props) => {
     showError && 'react-select--error',
   ].filter(Boolean).join(' ');
 
+  if (!isCreatable) {
+    return (
+      <Select
+        isLoading={isLoading}
+        placeholder={getTranslation(placeholder, i18n)}
+        captureMenuScroll
+        {...props}
+        value={value}
+        onChange={onChange}
+        isDisabled={disabled}
+        className={classes}
+        classNamePrefix="rs"
+        options={options}
+        isSearchable={isSearchable}
+        isClearable={isClearable}
+        filterOption={filterOption}
+        onMenuOpen={onMenuOpen}
+        menuPlacement="auto"
+        components={{
+          ValueContainer,
+          SingleValue,
+          MultiValue,
+          MultiValueLabel,
+          MultiValueRemove,
+          DropdownIndicator: Chevron,
+          ClearIndicator,
+          Control,
+          ...components,
+        }}
+      />
+    );
+  }
+  const handleKeyDown: KeyboardEventHandler = (event) => {
+    if (!value || !inputValue || inputValue.trim() === '') return;
+    switch (event.key) {
+      case 'Enter':
+      case 'Tab':
+        onChange([...value as Option[], createOption(inputValue)]);
```

Comment:
When I type something invalid and press tab, it will enter "NaN". Ideally, you wouldn't even be able to enter any string and the input would behave like a `type="number"`. Is that possible with react-select the way we are using it?

It wasn't clear in their docs if this is possible: https://react-select.com/components#:~:text=Yellow-,Input,-Input%20to%20render